metadata_backend/api/handlers.py

```diff
@@ -461,6 +463,32 @@ async def delete_user(self, req: Request) -> Response:
         LOG.info(f"DELETE user with ID {user} was successful.")
         return web.Response(status=204)
 
+    async def mirror_dataset(self, req: Request) -> Response:
+        """Mirror dataset content queried from EGA into the database.
+
+        :param req: POST request
+        :returns: HTTP No Content response
+        """
+        dataset_id = req.match_info["datasetId"]
+        db_client = req.app["db_client"]
+        await spawn(req, self.data_mirroring(dataset_id, db_client))
+        return web.Response(status=202)
+
+    async def data_mirroring(self, dataset_id: str, db_client: AsyncIOMotorClient) -> None:
+        """Get the dataset and its objects from EGA and POST the metadata objects.
+
+        :param dataset_id: ID of dataset in EGA
+        :param db_client: Motor client used for database connections
+        """
+        ega_data = MetadataMirror().mirror_dataset(dataset_id)
```

Comment:
we need to check if the database already contains the dataset and if it is in a published folder (because we are mirroring public information, I think the folder should be published by default, but if there is another option fine with that as well) so that we don't mirror it again

Reply:
Lines 890-892 of `handlers.py` has an attempt at doing this check. However, the check is not implemented properly yet and needs refining.